### ISSUE TITLE
vo=gpu opens regular MPV alongside memento

### ISSUE THREAD

**Issue by eyalmazuz** [ripose-jp/Memento#243]
I'm using [Anime4K](https://github.com/bloc97/Anime4K/) in MPV and I thought about adding it to memento as well
but when I run memento with the shaders (even when not activated by default)
I get the following error:
```
~/D/J/A/[Judas] Kimetsu no Yaiba - Mugen Ressha-hen [1080p][HEVC x265 10bit][Multi-Subs]> memento \[Judas\]\ Kimetsu\ no\ Yaiba\ -\ Mugen\ Ressha-hen\ -\ 05.mkv
AO: [pipewire] 44100Hz stereo 2ch floatp
VO: [libmpv] 1920x1080 yuv420p10
[libmpv_render] after creating texture: OpenGL error INVALID_ENUM.
[mpv_thumbnail_script_client_osc] The 'tick' event is deprecated and will be removed.
[libmpv_render] before loading program: OpenGL error INVALID_ENUM.
```

Which is an error by its own.
But to fix this I can add to the mpv.conf ``vo=gpu``

so now my config file looks like this:
```
keep-open=yes
osc=no
script-opts-add=osc-title=${media-title}${?chapter-metadata/title: [${chapter-metadata/title}]}
save-position-on-quit
vo=gpu
profile=gpu-hq
```

but when doing so, when opening with the new config file I get memento and MPV open at the same time, the video is displayed on the MPV screen and the subtitles on memento

![image](https://github.com/user-attachments/assets/26febc24-a972-4f50-af7d-acf52cdbea5f)

I can freely use the shaders now and clear then, but everything happens on the new MPV screen instead of having one instance of memento

This is unrelated to the shaders and scripts I have since I tried removing everything and the bug still persists

**Comment by ripose-jp** [comment]
Looks like it's a problem with Anime4K being incompatible with `vo=libmpv`. I'd report this to either mpv or Anime4K and get their opinion. There's not anything I can do to fix this.

>or is there a plan for future support of other --vo= options?

This will never happen because Memento uses libmpv and `vo=libmpv` is the only option for doing that.

**Comment by eyalmazuz** [comment]
> Looks like it's a problem with Anime4K being incompatible with `vo=libmpv`. I'd report this to either mpv or Anime4K and get their opinion. There's not anything I can do to fix this.
> 
> > or is there a plan for future support of other --vo= options?
> 
> This will never happen because Memento uses libmpv and `vo=libmpv` is the only supported `vo`.

I tried updating to the latest memento-main
and I'm getting the OpenGL error without any script/shaders added
on a very clean install

I'm suspecting it might not be a anime4K issue(?)

**Comment by Jefffrey** [comment]
I'm also observing this error on a fresh clone, build & install of this repo; even without Anime4k shaders installed. Running on Manjaro Linux x86_64.

**Comment by ripose-jp** [comment]
I can verify that it's happening on mpv v0.38.0 without any shaders. Looks like a bug in `vo=libmpv`. I'll try to collect more information and make a report to mpv.

```
[libmpv_render] Trying to load shader from disk...
[libmpv_render] before loading program: OpenGL error INVALID_ENUM.
```

```
[libmpv_render] Window size: 1920x1080 (Borders: l=0 t=0 r=0 b=0)
[libmpv_render] Video source: 1920x1080 (1:1)
[libmpv_render] Video display: (0, 0) 1920x1080 -> (0, 0) 1920x1080
[libmpv_render] Video scale: 1.000000/1.000000
[libmpv_render] OSD borders: l=0 t=0 r=0 b=0
[libmpv_render] Video borders: l=0 t=0 r=0 b=0
[libmpv_render] Texture for plane 0: 1920x1080
[libmpv_render] after creating texture: OpenGL error INVALID_ENUM.
[libmpv_render] Texture for plane 1: 960x540
[libmpv_render] Texture for plane 2: 960x540
[libmpv_render] Testing FBO format rgba16f
```

**Comment by ripose-jp** [comment]
This issue is being tracked at https://github.com/mpv-player/mpv/issues/15019

**Comment by eyalmazuz** [comment]
Update: with MPV 0.39 (?)

suddenly anime4k works without issues
had to change the shortcuts to ctrl+3-9 instead of ctrl+1-6 + ctrl+0

cause apparently memento uses ctrl+1 and ctrl+2 for subtitle sizes and ctrl+0 to hide/display subtitles

but running memento with the shaders, I can switch between and clear the shaders without issues (at least it looks like it's working)

I still get 

```
[libmpv_render] after creating texture: OpenGL error INVALID_ENUM.
[libmpv_render] before loading program: OpenGL error INVALID_ENUM.
```

when the video loads for the first time 
and 
```
[libmpv_render] after creating texture: OpenGL error INVALID_ENUM.
```

after every switch of shaders

but I can see changes in the image quality regardless

**Comment by eyalmazuz** [comment]
Since Memento 2.0.0 this issue doesn't appear anymore

```
memento ~/Desktop/japanese/anime/\[Erai-raws\]\ Hyakuemu\ -\ Movie\ \[1080p\ NF\ WEBRip\ HEVC\ EAC3\]\[MultiSub\]\[FE0A2AA1\].mkv
qrc:/qt/qml/org/kde/breeze/TextArea.qml:92:9: Unable to assign AnkiFieldTextArea_QMLTYPE_181 to QQuickTextInput
● Video  --vid=1  --vlang=ja       '[Erai-raws]_HEVC_NF' (hevc 1920x1080 23.976 fps) [default]
● Audio  --aid=1  --alang=ja       '[Erai-raws]_EAC3_NF' (eac3 6ch 48000 Hz) [default]
● Subs   --sid=1  --slang=en       'NF' (ass) [default]
○ Subs   --sid=9  --slang=ja       'NF' (ass)
AO: [pipewire] 48000Hz 5.1(side) 6ch floatp
VO: [libmpv] 1920x1080 yuv420p10
(Paused) AV: 00:00:01 / 01:47:11 (0%) A-V:  0.000 Dropped: 7
```

Maybe it can be closed

**Comment by ripose-jp** [comment]
I doubt this issue was fixed by something that changed for v2.0.0, but you can close it if you want. I'll leave it up to you.

**Comment by eyalmazuz** [comment]
> I doubt this issue was fixed by something that changed for v2.0.0, but you can close it if you want. I'll leave it up to you.

Yeah, probably something with memento compiling against newer MPV version, I'll close this issue for now since I had the other person test it as well and nothing for him on MacOS
